### DISCUSSION TITLE
How playful should our bot be?

### DISCUSSION THREAD

**Discussion by DonaldKellett** [codewars/discord-bot#5] [General]
Currently, our official bot has some basic moderation features built in, but should we limit our bot to just moderation features? As an example, the [Xena Project](https://www.ma.imperial.ac.uk/~buzzard/xena/) Discord server (public invite not available) has some 12 bots, where one of them (`lean-bot`) is "homegrown" for running Lean code snippets, one of them ([MEE6](https://mee6.xyz/)) is presumably used for moderation (though I've not seen it in action), but the other 10 are just bots "for fun", e.g. a `counting` bot for playing a game of taking turns counting upwards, some 2 bots related to chess, etc.

If we decide to add some "game" features to our bot, here are a few ideas (that should not require connecting to the code runner):

- BF interpreter, possibly configurable with "command-line" options
- Lambda calculus evaluator ( @Kacarott might be interested in this after learning TypeScript ;-)
- ~~[Z3 automated theorem prover](https://z3prover.github.io/api/html/) integration~~ it doesn't have JavaScript bindings :-(

Some possible pros:

- More banter
- More community engagement
- Attract more people to use Codewars, by using the Discord server as an entry point

Possible cons:

- Interpreters built-in to the bot may stall bot for longer than desirable (what if there is an infinite loop?), and time limits only slightly alleviate this problem
- Our bot becomes "bloated", depending on how you look at it
- Too many "gaming features" in our bot (and the Codewars Discord server) may distract the community away from Codewars itself to the Discord server

What do you think? Feel free to leave your opinions below.

**Comment by kazk** [comment]
I'll _consider_ commands for fun as long as it gets used occasionally, easy to maintain, doesn't require a huge dependency, it's secure, and doesn't use much resource. I won't add interpreters if it can't guarantee resource limits (like max memory).

Connecting to CodeRunner is easy, and I already have a working code, but I'm still thinking about it. If I decide to add support, it'll be a limited version for small snippets (single file, much shorter timeout, output size, etc.).

**Comment by Rhys-Alexander** [comment]
I think the people using the discord bot will already be using Codewars. It seams the main bonus would be fun. Like Kazk said, adding fun commands may work as long as they fit the criteria he specified. Ultimately, if the goal is adding more playful fun to engage the community more, adding Easter eggs or playful features to the site would reach a much wider audience, accomplishing the goal much more effectively.
